lisa/sut_orchestrator/baremetal/build.py

```diff
@@ -79,23 +105,49 @@ def copy(self, sources_path: List[Path], files_map: List[FileSchema]) -> None:
                         )
                 self._log.debug(f"copy file {file} to {share_name}\\{file_name}")
 
-    def _find_matched_files(
-        self, sources_path: List[Path], files_map: List[FileSchema]
-    ) -> Dict[str, FileSchema]:
-        all_files = []
-        match_files: Dict[str, FileSchema] = {}
-        for source_path in sources_path:
-            for root, _, files in os.walk(source_path):
-                for file in files:
-                    all_files.append(os.path.join(root, file))
-
-            for file_map in files_map:
-                file_path = rf"{source_path}\{file_map.source}".replace("\\", "\\\\")
-                pattern = re.compile(
-                    file_path,
-                    re.I | re.M,
-                )
-                for file in all_files:
-                    if pattern.match(file):
-                        match_files[file] = file_map
-        return match_files
+
+class TftpBuild(Build):
+    def __init__(self, runbook: TftpBuildSchema) -> None:
+        super().__init__(runbook)
+        self.pxe_runbook: TftpBuildSchema = self.runbook
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "tftp"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return TftpBuildSchema
+
+    def copy(self, sources_path: List[Path], files_map: List[FileSchema]) -> None:
+        assert self.pxe_runbook.connection, "The build server is not specified"
+
+        try:
+            build_server = quick_connect(
+                self.pxe_runbook.connection,
+                logger_name="build_server",
+            )
+        except Exception:
+            raise LisaException(f"Failed to connect {self.pxe_runbook.connection}")
+
+        ls = build_server.tools[Ls]
+        rc = build_server.tools[RemoteCopy]
+        for file, file_map in _find_matched_files(
+            sources_path,
+            files_map,
+        ).items():
+            if file_map.destination:
+                file_map_path = PurePath(file_map.destination)
+                if ls.is_file(file_map_path):
+                    file_destination = file_map_path.parent
+                else:
+                    file_destination = file_map_path
+            else:
+                file_destination = PurePath(file).parent
+
+            try:
+                rc.copy_to_remote(PurePath(file), file_destination)
+            except Exception:
```

Comment:
Similar as above, the try/except is not needed.

Reply:
Same as above. thanks